frontends/concrete-python/concrete/fhe/compilation/compiler.py

```diff
@@ -479,56 +487,123 @@ def compile(
+                    print("Computation Graph")
+                    print("-" * columns)
+                    print(self.graph.format())
+                    print("-" * columns)
 
-                try:  # pragma: no cover
-                    # this branch cannot be covered
-                    # because `os.get_terminal_size()`
-                    # raises an exception during tests
+                    print()
 
-                    columns, _ = os.get_terminal_size()
-                    if columns == 0:  # noqa: SIM108
-                        columns = min(longest_line, 80)
-                    else:
-                        columns = min(longest_line, columns)
-                except OSError:  # pragma: no cover
-                    columns = min(longest_line, 80)
+            # in-memory MLIR module
+            mlir_context = self.compilation_context.mlir_context()
+            mlir_module = GraphConverter().convert(self.graph, self.configuration, mlir_context)
+            # textual representation of the MLIR module
+            mlir_str = str(mlir_module).strip()
+            if self.artifacts is not None:
+                self.artifacts.add_mlir_to_compile(mlir_str)
 
-                if show_graph:
+            if show_bit_width_constraints:
+                if is_first:  # pragma: no cover
                     print()
+                    is_first = False
 
-                    print("Computation Graph")
-                    print("-" * columns)
-                    print(graph)
-                    print("-" * columns)
+                print("Bit-Width Constraints")
+                print("-" * columns)
+                for i, node in enumerate(nx.lexicographical_topological_sort(self.graph.graph)):
+                    if len(node.bit_width_constraints) > 0:
+                        print(f"%{i}:")
+                        for constraint in node.bit_width_constraints:
+                            print(
+                                f"    {constraint.arg(0)} {constraint.decl()} {constraint.arg(1)}"
+                            )
+                print("-" * columns)
+
+                print()
 
+            if show_bit_width_assignments:
+                if is_first:  # pragma: no cover
                     print()
+                    is_first = False
 
-                if show_mlir:
-                    print("\n" if not show_graph else "", end="")
+                print("Bit-Width Assignments")
+                print("-" * columns)
 
-                    print("MLIR")
-                    print("-" * columns)
-                    print(mlir_str)
-                    print("-" * columns)
+                lines = []
+                for variable in self.graph.bit_width_assignments.decls():  # type: ignore
+                    width = self.graph.bit_width_assignments.get_interp(variable)  # type: ignore
+                    lines.append(f"{variable} = {width}")
+
+                def sorter(line: str) -> int:
```

Comment:
should be a separate function printing the lines, this long  function doesn't need that orthogonal logic of line formatting inside.

Reply:
Done.